components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/DefaultSessionManager.cs

```diff
@@ -280,45 +323,36 @@ public class DefaultSessionManager : ISessionManager {
         /// </summary>
         /// <param name="session"></param>
         /// <param name="e"></param>
-        private void Session_KeepAlive(Session session, KeepAliveEventArgs e) {
+        private async void Session_KeepAlive(Session session, KeepAliveEventArgs e) {
```

Comment:
Unless you can disable keep alive callbacks firing I would suggest in this case to just use Wait() as you do not want x tasks from being scheduled for every keep alive callback that happens while this task is executing.   

Similar to timers, where you want to stop the timer while the task is executing and restart when the task is done.   (To prevent x tasks queing up on every timer fire, while the thread pool is exhausted).

In general async void should be avoided as much as possible, but in callback case it is not possible.   But we need to make sure that exceptions are caught, or else the entire process is getting killed.

Reply:
done